miden/src/cli/run.rs

```diff
@@ -107,24 +122,62 @@ impl RunCmd {
 // HELPER FUNCTIONS
 // ================================================================================================
 
-#[instrument(name = "run_program", skip_all)]
-fn run_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Report> {
+#[instrument(name = "run_masp_program", skip_all)]
+fn run_masp_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Report> {
+    let bytes = fs::read(&params.program_file)
+        .into_diagnostic()
+        .wrap_err("Failed to read package file")?;
+
+    let package = Package::read_from_bytes(&bytes)
+        .into_diagnostic()
+        .wrap_err("Failed to deserialize package")?;
+
+    let program_arc: Arc<vm_core::Program> = match package.into_mast_artifact() {
+        MastArtifact::Executable(program_arc) => program_arc,
+        _ => return Err(Report::msg("The provided package is not a program package.")),
+    };
+    let program = &*program_arc;
+
+    // use simplified input data reading
+    let input_data = InputFile::read(&params.input_file, &params.program_file)?;
+
+    let stack_inputs = input_data.parse_stack_inputs().map_err(Report::msg)?;
+    let mut host = DefaultHost::default();
+
+    let execution_options = ExecutionOptions::new(
+        Some(params.max_cycles),
+        params.expected_cycles,
+        params.trace,
+        params.debug,
+    )
+    .into_diagnostic()?;
+
+    let program_hash: [u8; 32] = program.hash().into();
+
+    // execute program and generate outputs
+    let trace = processor::execute(program, stack_inputs, &mut host, execution_options)
+        .into_diagnostic()
+        .wrap_err("Failed to generate execution trace")?;
+
+    Ok((trace, program_hash))
+}
+
+#[instrument(name = "run_masp_program", skip_all)]
+fn run_masm_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Report> {
```

Comment:
The `name` attribute here should be "run_masm_pogram". But also, we could probably just use "run_program" both here and on line 125 above (i.e., the names of functions would stay as they are now, but we'd use the same name for the `instrument` attribute).